### PR TITLE
add select2 to user impersonation and shortlink edit

### LINKED ISSUE #537: Select2
use select2 for view as user selection and shortlink document selection

### PR DESCRIPTION
**Pull request by Bartzi**
fixes #537

### CODE REVIEW

**Comment by janno42** [comment]
please make sure the internal link select2 still works after this pr

File: _1327/shortlinks/forms.py [hunk 1]
```diff
@@ -11,6 +11,10 @@ class Meta:
 		model = Shortlink
 		fields = ("url_title", "link", "document")
 
+	def __init__(self, *args, **kwargs):
+		super(ShortlinkForm, self).__init__(*args, **kwargs)
+		self.fields['document'].widget.attrs['class'] = 'select2-selection'
```
Comment (janno42) [inline _1327/shortlinks/forms.py]: please group this by document type like in the internal link select2

**Comment by Bartzi** [comment]
fixed it, please review again as I had to change quite a lot

File: _1327/templates/base.html [hunk 2]
```diff
@@ -240,7 +240,8 @@
 				$(this).parent().addClass('menu-hover');
 			}).on('mouseout', function(){
 				$(this).parent().removeClass('menu-hover');
-			})
+			});
+
```
Comment (janno42) [inline _1327/templates/base.html]: why?

**Comment by janno42** [comment]
there is still no select2 on the user impersonation page because select2 is not called on the form field in jacascript on this specific page yet.

**Comment by Bartzi** [comment]
Hmm I thought it works now... I'll fix it ;)

**Pull request merged**